### ISSUE TITLE
pip install torch-scatter WRONG! error: ‘torch::autograd::AutogradContext’ has not been declared?

### ISSUE THREAD

**Issue by zeusbrav** [rusty1s/pytorch_geometric#1250]
## ❓ Questions & Help

ERROR: Command errored out with exit status 1:
   command: /usr/bin/python3 -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-vmou20t4/torch-scatter/setup.py'"'"'; __file__='"'"'/tmp/pip-install-vmou20t4/torch-scatter/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-z7mjlp9x
       cwd: /tmp/pip-install-vmou20t4/torch-scatter/
  Complete output (36 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.linux-x86_64-3.6
  creating build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/__init__.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/placeholder.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/scatter.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/segment_coo.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/segment_csr.py -> build/lib.linux-x86_64-3.6/torch_scatter
  copying torch_scatter/utils.py -> build/lib.linux-x86_64-3.6/torch_scatter
  creating build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/__init__.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/logsumexp.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/softmax.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  copying torch_scatter/composite/std.py -> build/lib.linux-x86_64-3.6/torch_scatter/composite
  running build_ext
  building 'torch_scatter._scatter' extension
  creating build/temp.linux-x86_64-3.6
  creating build/temp.linux-x86_64-3.6/tmp
  creating build/temp.linux-x86_64-3.6/tmp/pip-install-vmou20t4
  creating build/temp.linux-x86_64-3.6/tmp/pip-install-vmou20t4/torch-scatter
  creating build/temp.linux-x86_64-3.6/tmp/pip-install-vmou20t4/torch-scatter/csrc
  creating build/temp.linux-x86_64-3.6/tmp/pip-install-vmou20t4/torch-scatter/csrc/cpu
  creating build/temp.linux-x86_64-3.6/tmp/pip-install-vmou20t4/torch-scatter/csrc/cuda
  x86_64-linux-gnu-gcc -pthread -DNDEBUG -g -fwrapv -O2 -Wall -g -fstack-protector-strong -Wformat -Werror=format-security -Wdate-time -D_FORTIFY_SOURCE=2 -fPIC -DWITH_CUDA -I/tmp/pip-install-vmou20t4/torch-scatter/csrc -I/usr/local/lib/python3.6/dist-packages/torch/include -I/usr/local/lib/python3.6/dist-packages/torch/include/torch/csrc/api/include -I/usr/local/lib/python3.6/dist-packages/torch/include/TH -I/usr/local/lib/python3.6/dist-packages/torch/include/THC -I/usr/local/cuda/include -I/usr/include/python3.6m -c /tmp/pip-install-vmou20t4/torch-scatter/csrc/scatter.cpp -o build/temp.linux-x86_64-3.6/tmp/pip-install-vmou20t4/torch-scatter/csrc/scatter.o -DTORCH_API_INCLUDE_EXTENSION_H -DTORCH_EXTENSION_NAME=_scatter -D_GLIBCXX_USE_CXX11_ABI=0 -std=c++11
  /tmp/pip-install-vmou20t4/torch-scatter/csrc/scatter.cpp:40:24: error: ‘torch::autograd::AutogradContext’ has not been declared
   using torch::autograd::AutogradContext;
                          ^
  /tmp/pip-install-vmou20t4/torch-scatter/csrc/scatter.cpp:44:52: error: expected template-name before ‘<’ token
   class ScatterSum : public torch::autograd::Function<ScatterSum> {
                                                      ^
  /tmp/pip-install-vmou20t4/torch-scatter/csrc/scatter.cpp:44:52: error: expected ‘{’ before ‘<’ token
  /tmp/pip-install-vmou20t4/torch-scatter/csrc/scatter.cpp:44:52: error: expected unqualified-id before ‘<’ token
  error: command 'x86_64-linux-gnu-gcc' failed with exit status 1

**Comment by rusty1s** [comment]
That looks like you use an older version than PyTorch 1.4. Is that correct?

**Comment by zeusbrav** [comment]
yes, you are right.I am using pytorch 1.2.0. The cuda is 10.0 and I will try to use another pytorch version. Do you have any advice?  Thx

**Comment by zeusbrav** [comment]
> That looks like you use an older version than PyTorch 1.4. Is that correct?

yes, you are right. I am using pytorch 1.2.0. The cuda is 10.0 and I will try to use another pytorch version. Do you have any advice?  Thx

**Comment by rusty1s** [comment]
We only support PyTorch >= 1.4, so upgrading to a newer version should definitely resolve this.
